src/Elastic.Apm/Elastic.Apm.csproj

```diff
@@ -22,9 +22,8 @@
     <LangVersion>latest</LangVersion>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
-    <!-- TODO: Are we ok with this reference/version? -->
-    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="4.0.0" />
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.2" />
+    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="4.5.1" />
```

Comment:
If you're referring to `System.Diagnostics.DiagnosticSource`, are you sure, because `Elastic.Apm.EntityFrameworkCore.csproj` already refers to 4.5.1 (_before_ my change)? And I think this is where I was having issues because of the disparate versions. I consolidated them to the latest of the versions referenced.

Reply:
Och, I see.. that's bad - and probably my mistake :) We should prefer the older version. So we should have `4.0.0` in `Elastic.Apm.EntityFrameworkCore.csproj`

Reply:
That doesn't appear possible because of the reference to Microsoft.EntityFrameworkCore 2.0.0. This requires System.Diagnostics.DiagnosticSource >= 4.4.1 as per the build error I get:

> Detected package downgrade: System.Diagnostics.DiagnosticSource from 4.4.1 to 4.0.0. Reference the package directly from the project to select a different version. 
>  Elastic.Apm.EntityFrameworkCore -> Microsoft.EntityFrameworkCore.Relational 2.0.0 -> Microsoft.EntityFrameworkCore 2.0.0 -> System.Diagnostics.DiagnosticSource (>= 4.4.1) 
>  Elastic.Apm.EntityFrameworkCore -> System.Diagnostics.DiagnosticSource (>= 4.0.0)	Elastic.Apm.EntityFrameworkCore

I seem to recall System.Diagnostics.DiagnosticSource was what was causing me run-time problems in my client project to do with this package. That may have been why I did the canvas upgrade. So, I think there's no real option other than to use at least 4.4.1, in which case may as well use 4.5.1.

Reply:
I see. Makes sense, yeah, this is not right and I see why you ran into this problem.

This dependency is unnecessary. You can remove it, since it gets pulled through `Microsoft.EntityFrameworkCore.Relational`, it's version does not matter... so if `Microsoft.EntityFrameworkCore.Relational` pulls a newer version for the `Elastic.Apm.EntityFrameworkCore` then it's fine, with that our `Elastic.Apm` project still targets older versions.

It's btw. the same with the `Elastic.Apm.AspNetCore` project: that also depends on `DiagnosticSource`, but we don't reference it directly, it comes as a dependency of one of the other dependencies. No idea why I added it to the `Elastic.Apm.EntityFrameworkCore` project.

So... in short: just remove that line with `System.Diagnostics.DiagnosticSource` from the `.csproj` file. It should work, just tried it.

Reply:
Great, just done that and committed.